File: solutions/2088-minimum-time-to-type-word-using-special-typewriter/minimum-time-to-type-word-using-special-typewriter.cpp
class Solution {
public:
    int minTimeToType(string word) {
        int ans=word.size(); 
        int cur_ptr=1; 
        for(char c : word)
        {
            int pos=c-96; 
            int diff=abs(pos-cur_ptr); 
            ans+=min(diff,26-diff);
            cur_ptr=pos; 
        }
        return ans;
    }
};

File: solutions/2088-minimum-time-to-type-word-using-special-typewriter/minimum-time-to-type-word-using-special-typewriter.rs
impl Solution {
    pub fn min_time_to_type(word: String) -> i32 {
        let mut ans = word.len() as i32;
        let mut cur = 1i32;
        for b in word.bytes() {
            let pos = (b - b'a' + 1) as i32;
            let diff = (pos - cur).abs();
            ans += diff.min(26 - diff);
            cur = pos;
        }
        ans
    }
}